### PR TITLE
feat(translation): add German translations

### PR DESCRIPTION
**Pull request by Rishabh04-02**
Issue - Translation for other languages #439

### CODE REVIEW

**Comment by Rishabh04-02** [comment]
@housseindjirdeh @andrewda Please review.

File: src/auth/screens/language-settings.js [hunk 1]
```diff
@@ -29,4 +29,9 @@ export default [
     emojiCode: ':flag-ru:',
     name: 'Русский',
   },
+  {
+    code: 'de',
+    emojiCode: ':flag-de:',
+    name: 'Deutsche',
```
Comment (lex111) [inline src/auth/screens/language-settings.js]: Why not `Deutsch`?

File: src/locale/languages/de.js [hunk 2]
```diff
@@ -0,0 +1,335 @@
+export const de = {
+  auth: {
+    login: {
+      connectingToGitHub: 'Verbindung zu GitHub herstellen ...',
+      preparingGitPoint: 'GitPoint vorbereiten ...',
```
Comment (lex111) [inline src/locale/languages/de.js]: Space is precisely necessary?

File: src/auth/screens/language-settings.js (hunk 1 above)
Comment (Rishabh04-02) [reply src/auth/screens/language-settings.js]: as i was shown this result [here](https://translate.google.com/#en/de/german)
Should i change it to ``Deutsch``?

File: src/locale/languages/de.js (hunk 2 above)
Comment (Rishabh04-02) [reply src/locale/languages/de.js]: No, its not necessary. I'll change it.

File: src/locale/languages/de.js [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+        closed: 'abgeschlossen',
+        reopened: 'wiedereröffnet',
+        review_requested: 'Überprüfung angefordert',
+        review_request_removed: 'Überprüfungsanforderung entfernt',
+        submitted: 'eingereicht',
+        dismissed: 'entlassen',
+        published: 'veröffentlicht',
+        publicized: 'publiziert',
+        privatized: 'privatisiert',
+        starred: 'Stern',
```
Comment (lex111) [inline src/locale/languages/de.js]: Is this exactly true? "starred" a verb, but `Stern` a noun.

Comment (Rishabh04-02) [reply src/locale/languages/de.js]: Yaa, i'll change it to starred

File: src/auth/screens/language-settings.js (hunk 1 above)
Comment (lex111) [reply src/auth/screens/language-settings.js]: Yes, because we use the adjective.

`Deutsch` - adjective
`Deutsche` - noun

**Comment by lex111** [comment]
@Rishabh04-02 German is your native language?

**Comment by Rishabh04-02** [comment]
@lex111 Please review.

File: src/locale/languages/de.js [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+        closed: 'abgeschlossen',
+        reopened: 'wiedereröffnet',
+        review_requested: 'Überprüfung angefordert',
+        review_request_removed: 'Überprüfungsanforderung entfernt',
+        submitted: 'eingereicht',
+        dismissed: 'entlassen',
+        published: 'veröffentlicht',
+        publicized: 'publiziert',
+        privatized: 'privatisiert',
+        starred: 'Starred',
```
Comment (lex111) [inline src/locale/languages/de.js]: Must start with a small letter. And why `starred`, in German is how will it be? `bemerken`?

**Comment by Rishabh04-02** [comment]
> German is your native language?
No, Its not.

File: src/locale/languages/de.js (hunk 4 above)
Comment (Rishabh04-02) [reply src/locale/languages/de.js]: I'm not sure about it, here it is showing that ``bemerken`` translates to - ``to notice`` in english [here](https://translate.google.com/#de/en/bemerken)

So, should i only convert it to lowercase?

**Comment by lex111** [comment]
@Rishabh04-02 thanks for the help, although it was not written in the issue #439, but it is desirable, of course, that the translation was from the native speaker, since then it will be more accurate. Because the translation with Google Translate sometimes gives unexpected results.

@housseindjirdeh what should we do in this case? Waiting for the native translator to check this translation?

**Comment by Rishabh04-02** [comment]
@lex111 Yaa, I can understand that It's important to validate the content from native speaker 👍.
But please mention this requirement in the issue #439 , as It is possible that few more developers are doing the same thing as I did and not knowing this initially will waste their time and their efforts will go in vain.

**Comment by FlorianTaut** [comment]
I can review, german is my native language @lex111

**Comment by lex111** [comment]
@FlorianTaut ok, that was cool!

**Comment by FlorianTaut** [comment]
@Rishabh04-02 did you translate this with google translator? Because I just read some lines and they make no sense.

**Comment by Rishabh04-02** [comment]
@FlorianTaut 
> @Rishabh04-02 did you translate this with google translator? Because I just read some lines and they make no sense.

Yes, I've used google translator for most of the translations and the English to German dictionary for the others.

**Comment by FlorianTaut** [comment]
The problem is, this translation is nearly unuseful. 90% are wrong translated and make no sense. 

How should we go on with this?

There is so much to do that in my opinion it would be more useful to delete this pr an make a complete new one. I already have it translated but I’m currently not possible to test it in the emulator so I didn’t sent a pr yet.

**Comment by lex111** [comment]
Well done, if this given translation is almost completely incorrect, and there is a good translation, I agree, it makes no sense to edit the current PR and it should be closed then.

@FlorianTaut you have previously sent a PR with a German translation and closed because it was not possible to check? Well, if the translation was finished, then you did _not_ need to close the PR, because In any case, this is a fixable problem, after all, someone else has checked that the application is working with a new translation.

In general, if there is a translation, send PR, the main thing is to have the correct translation. If something does not work, then you can write to Gitter, we will try to help you.

**Comment by Rishabh04-02** [comment]
@FlorianTaut Thanks for reviewing, I've closed the PR now.

**Comment by lex111** [comment]
@Rishabh04-02 thanks anyway for wanting to help. If you know React Native, and you have a desire to help, then you can take another [issue](https://github.com/gitpoint/git-point/issues?q=is%3Aopen+is%3Aissue+label%3Ahacktoberfest), we will be glad and will try to help if there are difficulties.

**Comment by housseindjirdeh** [comment]
Thank you for this @Rishabh04-02 and apologies on my part, I should have been more clear in the contributing docs that we're looking for language speakers to translate our content in depth. I pushed a [commit](https://github.com/gitpoint/git-point/commit/20327b8748328e2c82b252a76aa3bf28a15f47ef) to make things more clear.

Really appreciate the time you took to translate each string and please don't hesitate to snag any issue you think you'd like to help with as @lex111 mentioned.

**Comment by Rishabh04-02** [comment]
@housseindjirdeh @lex111 Thanks for the steps you took to avoid such problems in future. I'll be looking forward to contribute to this awesome project 😄 .

**Comment by housseindjirdeh** [comment]
Thank you so much mate, can't tell you how much we appreciate that :tada: :tada: :tada:

**Pull request closed without merging**